File: 003)methodsAndClasses/app/src/main/java/com/example/a003methodsandclasses/Family.java
package com.example.a003methodsandclasses;

public class Family {
    private String name;
    private int age;

    public Family() {
    }

    public Family(String name, int age) {
        this.name = name;
        this.age = age;
        System.out.println("constructor called");
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}

File: 054)KotlinOOPProject/app/src/main/java/com/elyesasimsek/kotlinoopproject/Family.kt
package com.elyesasimsek.kotlinoopproject

open class Family(private var name: String?, private var age: Int?) {
    fun getName(): String? {
        return name
    }

    fun setName(newName: String?) {
        // name özelliği val ile tanımlandığı için değiştirilemez.
        // Eğer name özelliğini değiştirmek istiyorsanız, var ile tanımlamanız gerekir.
        // Örneğin: private var name: String?
        // Bu durumda, setName metodu şu şekilde olabilir:
         name = newName
    }

    fun getAge(): Int? {
        return age
    }

    fun setAge(newAge: Int?) {
        // age özelliği valile tanımlandığı için değiştirilemez.
        // Eğer age özelliğini değiştirmek istiyorsanız, var ile tanımlamanız gerekir.
        // Örneğin: private var age: Int?
        // Bu durumda, setAge metodu şu şekilde olabilir:
         age = newAge
    }

}